File: fix_luxora.py
import os
import re

directory = './'
files = [f for f in os.listdir(directory) if f.endswith('.html')]

footer_services_old = re.compile(
    r'<ul class=\"space-y-2\">\s*<li><a[^>]*href=\"#\"[^>]*>Strategy</a></li>\s*<li><a[^>]*href=\"#\"[^>]*>Editorial</a></li>\s*<li><a[^>]*href=\"#\"[^>]*>Identity</a></li>\s*</ul>',
    re.IGNORECASE
)

footer_services_fallback = re.compile(
    r'>Strategy</a></li>\s*<li>[^<]*<a[^>]*>Editorial</a></li>\s*<li>[^<]*<a[^>]*>Identity</a></li>',
    re.IGNORECASE
)

footer_services_new = '''<ul class="space-y-2">
          <li><a class="text-stone-400 dark:text-zinc-500 hover:text-amber-700 dark:hover:text-amber-500 transition-all font-sans uppercase tracking-[0.15em] text-[10px]" href="/luxury-fashion-brands.html">Luxury Fashion</a></li>
          <li><a class="text-stone-400 dark:text-zinc-500 hover:text-amber-700 dark:hover:text-amber-500 transition-all font-sans uppercase tracking-[0.15em] text-[10px]" href="/designer-boutiques.html">Designer Boutiques</a></li>
          <li><a class="text-stone-400 dark:text-zinc-500 hover:text-amber-700 dark:hover:text-amber-500 transition-all font-sans uppercase tracking-[0.15em] text-[10px]" href="/bespoke-tailoring.html">Bespoke Tailoring</a></li>
          <li><a class="text-stone-400 dark:text-zinc-500 hover:text-amber-700 dark:hover:text-amber-500 transition-all font-sans uppercase tracking-[0.15em] text-[10px]" href="/couture-wedding-wear.html">Couture & Wedding</a></li>
          <li><a class="text-stone-400 dark:text-zinc-500 hover:text-amber-700 dark:hover:text-amber-500 transition-all font-sans uppercase tracking-[0.15em] text-[10px]" href="/premium-ethnic-wear.html">Premium Ethnic Wear</a></li>
        </ul>'''

footer_services_new_fallback = '>Luxury Fashion</a></li><li><a class="text-stone-400 dark:text-zinc-500 hover:text-amber-700 dark:hover:text-amber-500 transition-all font-sans uppercase tracking-[0.15em] text-[10px]" href="/designer-boutiques.html">Designer Boutiques</a></li><li><a class="text-stone-400 dark:text-zinc-500 hover:text-amber-700 dark:hover:text-amber-500 transition-all font-sans uppercase tracking-[0.15em] text-[10px]" href="/bespoke-tailoring.html">Bespoke Tailoring</a></li><li><a class="text-stone-400 dark:text-zinc-500 hover:text-amber-700 dark:hover:text-amber-500 transition-all font-sans uppercase tracking-[0.15em] text-[10px]" href="/couture-wedding-wear.html">Couture & Wedding</a></li><li><a class="text-stone-400 dark:text-zinc-500 hover:text-amber-700 dark:hover:text-amber-500 transition-all font-sans uppercase tracking-[0.15em] text-[10px]" href="/premium-ethnic-wear.html">Premium Ethnic Wear</a></li>'

for f in files:
    filepath = os.path.join(directory, f)
    with open(filepath, 'r', encoding='utf-8') as file:
        content = file.read()

    # Footer replacements
    content = footer_services_old.sub(footer_services_new, content)
    content = footer_services_fallback.sub(footer_services_new_fallback, content)

    if f == 'index.html':
        content = content.replace('<section class="py-24 bg-primary overflow-hidden">', '<section class="py-24 bg-surface-container-low overflow-hidden">')
        content = content.replace('<div class="text-white text-center md:text-left mt-8 max-w-2xl px-2">', '<div class="text-primary text-center md:text-left mt-8 max-w-2xl px-2">')

    # Image fixes
    content = content.replace('young-woman-with-her-beautiful-wedding-dress.jpg', 'woman-robe-siitng-armchair-looking-wedding-dress-wall.jpg')
    content = content.replace('two-young-women-working-clothing-store (1).jpg', 'two-young-women-working-clothing-store.jpg')
    content = content.replace('two-young-women-working-clothing-store%20(1).jpg', 'two-young-women-working-clothing-store.jpg')
    content = content.replace('good-teamwork-is-key-success (1).jpg', 'good-teamwork-is-key-success.jpg')
    content = content.replace('good-teamwork-is-key-success%20(1).jpg', 'good-teamwork-is-key-success.jpg')

    # Spacing fixes
    content = content.replace('py-48', 'py-16')
    content = content.replace('pb-32', 'pb-12')
    content = content.replace('mb-32', 'mb-12')

    with open(filepath, 'w', encoding='utf-8') as file:
        file.write(content)
    print(f"Processed {f}")

# Rename images
public_dir = './public'
if os.path.exists(public_dir):
    for img in os.listdir(public_dir):
        if '(1)' in img or '%20(1)' in img:
            old_path = os.path.join(public_dir, img)
            new_name = img.replace(' (1)', '').replace('%20(1)', '')
            new_path = os.path.join(public_dir, new_name)
            os.rename(old_path, new_path)
            print(f"Renamed {img} to {new_name}")

File: fix_luxora.js
const fs = require('fs');
const path = require('path');

const dir = './';
const files = fs.readdirSync(dir).filter(f => f.endsWith('.html'));

const footerServicesOld = `<ul class="space-y-2">
          <li><a
              class="text-stone-400 dark:text-zinc-500 hover:text-amber-700 dark:hover:text-amber-500 transition-all font-sans uppercase tracking-[0.15em] text-[10px]"
              href="#">Strategy</a></li>
          <li><a
              class="text-stone-400 dark:text-zinc-500 hover:text-amber-700 dark:hover:text-amber-500 transition-all font-sans uppercase tracking-[0.15em] text-[10px]"
              href="#">Editorial</a></li>
          <li><a
              class="text-stone-400 dark:text-zinc-500 hover:text-amber-700 dark:hover:text-amber-500 transition-all font-sans uppercase tracking-[0.15em] text-[10px]"
              href="#">Identity</a></li>
        </ul>`;

const footerServicesNew = `<ul class="space-y-2">
          <li><a class="text-stone-400 dark:text-zinc-500 hover:text-amber-700 dark:hover:text-amber-500 transition-all font-sans uppercase tracking-[0.15em] text-[10px]" href="/luxury-fashion-brands.html">Luxury Fashion</a></li>
          <li><a class="text-stone-400 dark:text-zinc-500 hover:text-amber-700 dark:hover:text-amber-500 transition-all font-sans uppercase tracking-[0.15em] text-[10px]" href="/designer-boutiques.html">Designer Boutiques</a></li>
          <li><a class="text-stone-400 dark:text-zinc-500 hover:text-amber-700 dark:hover:text-amber-500 transition-all font-sans uppercase tracking-[0.15em] text-[10px]" href="/bespoke-tailoring.html">Bespoke Tailoring</a></li>
          <li><a class="text-stone-400 dark:text-zinc-500 hover:text-amber-700 dark:hover:text-amber-500 transition-all font-sans uppercase tracking-[0.15em] text-[10px]" href="/couture-wedding-wear.html">Couture & Wedding</a></li>
          <li><a class="text-stone-400 dark:text-zinc-500 hover:text-amber-700 dark:hover:text-amber-500 transition-all font-sans uppercase tracking-[0.15em] text-[10px]" href="/premium-ethnic-wear.html">Premium Ethnic Wear</a></li>
        </ul>`;

files.forEach(file => {
  const filePath = path.join(dir, file);
  let content = fs.readFileSync(filePath, 'utf8');

  // Fix footer services links
  // We use a regex that matches the ul class space-y-2 under Services in the footer
  const footerServicesRegex = /<ul class="space-y-2">\s*<li><a[^>]*href="#"[^>]*>Strategy<\/a><\/li>\s*<li><a[^>]*href="#"[^>]*>Editorial<\/a><\/li>\s*<li><a[^>]*href="#"[^>]*>Identity<\/a><\/li>\s*<\/ul>/i;
  content = content.replace(footerServicesRegex, footerServicesNew);
  
  // Alternative fallback if white space is different
  content = content.replace(/>Strategy<\/a><\/li>\s*<li>[^<]*<a[^>]*>Editorial<\/a><\/li>\s*<li>[^<]*<a[^>]*>Identity<\/a><\/li>/gi, 
                            '>Luxury Fashion</a></li><li><a class="text-stone-400 dark:text-zinc-500 hover:text-amber-700 dark:hover:text-amber-500 transition-all font-sans uppercase tracking-[0.15em] text-[10px]" href="/designer-boutiques.html">Designer Boutiques</a></li><li><a class="text-stone-400 dark:text-zinc-500 hover:text-amber-700 dark:hover:text-amber-500 transition-all font-sans uppercase tracking-[0.15em] text-[10px]" href="/bespoke-tailoring.html">Bespoke Tailoring</a></li><li><a class="text-stone-400 dark:text-zinc-500 hover:text-amber-700 dark:hover:text-amber-500 transition-all font-sans uppercase tracking-[0.15em] text-[10px]" href="/couture-wedding-wear.html">Couture & Wedding</a></li><li><a class="text-stone-400 dark:text-zinc-500 hover:text-amber-700 dark:hover:text-amber-500 transition-all font-sans uppercase tracking-[0.15em] text-[10px]" href="/premium-ethnic-wear.html">Premium Ethnic Wear</a></li>');

  // Index specific fixes
  if (file === 'index.html') {
    // Background offwhite for video section
    content = content.replace(/<section class="py-24 bg-primary overflow-hidden">/g, '<section class="py-24 bg-surface-container-low overflow-hidden">');
    content = content.replace(/<div class="text-white text-center md:text-left mt-8 max-w-2xl px-2">/g, '<div class="text-primary text-center md:text-left mt-8 max-w-2xl px-2">');
  }

  // Broken image fixes
  content = content.replace(/young-woman-with-her-beautiful-wedding-dress\.jpg/g, 'woman-robe-siitng-armchair-looking-wedding-dress-wall.jpg');
  content = content.replace(/two-young-women-working-clothing-store \(1\)\.jpg/g, 'two-young-women-working-clothing-store.jpg');
  content = content.replace(/two-young-women-working-clothing-store%20\(1\)\.jpg/g, 'two-young-women-working-clothing-store.jpg');
  content = content.replace(/good-teamwork-is-key-success \(1\)\.jpg/g, 'good-teamwork-is-key-success.jpg');
  content = content.replace(/good-teamwork-is-key-success%20\(1\)\.jpg/g, 'good-teamwork-is-key-success.jpg');

  // Fix padding to reduce negative space (maybe bottom elements had big pb or h-screen applied)
  content = content.replace(/py-48/g, 'py-16');
  content = content.replace(/pb-32/g, 'pb-12');
  content = content.replace(/mb-32/g, 'mb-12');
  
  content = content.replace(/h-\[100dvh\] overflow-y-auto w-full/g, 'h-[100dvh] overflow-y-auto w-[80%]'); // This was in fix_ios
  
  // Some pages had "min-h-[100dvh]" which can cause issues with padding. Let's make sure section paddings are balanced.
  
  fs.writeFileSync(filePath, content, 'utf8');
});

console.log('Fixed links, background, and broken image src. Moving on to rename images.');

// Rename images in public
const publicDir = './public';
if (fs.existsSync(publicDir)) {
  const images = fs.readdirSync(publicDir);
  images.forEach(img => {
    if (img.includes('(1)')) {
      const oldPath = path.join(publicDir, img);
      const newPath = path.join(publicDir, img.replace(' (1)', '').replace('%20(1)', ''));
      fs.renameSync(oldPath, newPath);
      console.log(`Renamed ${img} to ${path.basename(newPath)}`);
    }
  });
}
